### PR TITLE
Feature-gate the executable auto-download

### LINKED ISSUE #88: Make build-dep optional if not needed
If no automatic download feature is specified, deps like ureq should not be loaded

### PR DESCRIPTION
**Pull request by RCasatta**
This way all the build-deps are not used when auto-download is not required granting a MSRV of 1.41.1

While 1.57 is required if auto-download feature is used

closes #88

could help with #100 and #95

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -294,6 +294,8 @@ impl BitcoinD {
             p2p_args,
             conf_args
         );
+
+        #[allow(clippy::needless_borrow)] // would break 1.41.1 if fixed
```
Comment (Kixunil) [inline src/lib.rs:298]: FYI you can configure clippy to know about your MSRV and not suggest wrong things.

**Review by MaxFangX — APPROVED** [review]
tACK - Didn't squint too hard at the changes in `build.rs`, but I ran this patch and can confirm this PR fixes the dependency resolution issue described in #100, and also that all of our integration tests continue to work as expected.

**Comment by DanGould** [comment]
Seems like the right kind of thing to me. Very glad to see this addressed!

**Comment by DanGould** [comment]
MSRV seems impossibly confusing to maintain.

when I `cargo msrv` on this branch I get 1.54, which isn't even what you mention as required for all the features. Any clue what i'm doing wrong or how to figure out what is making the MSRV tick higher in my environment?

**Comment by RCasatta** [comment]
did https://github.com/Kixunil/payjoin/pull/40#issuecomment-1374897108 fix your issue?

**Comment by DanGould** [comment]
Not yet because I'm not sure how to figure out which particular dependency is causing MSRV to be higher than I'd like.

**Comment by Kixunil** [comment]
I do it by trial-error approach. Just try to compile it and whichever returns error I check the changelog and downgrade. It's a bit tedious but not too terrible.

**Pull request merged**